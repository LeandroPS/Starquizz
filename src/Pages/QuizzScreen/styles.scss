.quizz-screen{

    .header{
        padding: 20px 0px;

        img{
            height: 40px;
        }

        .timer{
            position: absolute;
            right: 0;
            top: 0;
        }
    }
    
    .cards-wrapper {
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
        grid-gap: 15px;

        padding: 20px 0px;
    }

    .navigation-buttons{
        position: relative;
        left: 50%;
        transform: translateX(-50%);
        margin-bottom: 20px;
    }
}