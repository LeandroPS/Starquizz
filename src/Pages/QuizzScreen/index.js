import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Card from '../../Components/Card';
import Timer from '../../Components/Timer';
import NavigationButtons from '../../Components/NavigationButtons';
import FeedbackWindow from '../../Components/FeedbackWindow';

import './styles.scss';
import '../../styles/common.scss';

class QuizzScreen extends Component{
    constructor(){
        super();

        this.state = {
            showFeedbackScreen: false,
            currentPage: 0,
            pointsCount: 0,
            guessedCharacters: []
        }
    }

    onTimeUp = () =>{
        this.setState({showFeedbackScreen: true});
    }

    includeGuessedCharacter = (characterName) =>{
        this.setState({guessedCharacters: [...this.state.guessedCharacters, characterName]})
    }

    addPoints = (points) =>{
        this.setState({pointsCount: this.state.pointsCount+points})
    }

    render(){
        const {characters, onNextPage, onPreviousPage, getResource} = this.props;
        const {guessedCharacters, pointsCount, showFeedbackScreen} = this.state;

        return(
            <React.Fragment>
                
                <div className="quizz-screen">
                    <div className="header">
                        <div className="container">
                            <img 
                                alt="CabeÃ§a do Darth Vader" 
                                src={require('../../assets/darth.png')}/>

                            <Timer initial={120} onTimeUp={this.onTimeUp} />
                        </div>
                    </div>
                    <div className="container">
                        <div className="cards-wrapper">
                            {characters.map((character, i) => {
                                return( 
                                    <Card 
                                        key={i}
                                        character={character} 
                                        guessed={guessedCharacters.indexOf(character.name)!==-1}
                                        includeGuessedCharacter={this.includeGuessedCharacter}
                                        addPoints={this.addPoints}
                                        getResource={getResource}/>);
                            })}    
                        </div>
                    </div>
                    <NavigationButtons onNextPage={onNextPage} onPreviousPage={onPreviousPage}/>
                    <FeedbackWindow points={pointsCount} show={showFeedbackScreen}/>
                </div>
            </React.Fragment>
        );
    }
}

QuizzScreen.propTypes = {
    characters: PropTypes.array.isRequired, 
    onNextPage: PropTypes.func.isRequired,
    onPreviousPage: PropTypes.func.isRequired,
    getResource: PropTypes.func.isRequired
}

export default QuizzScreen;