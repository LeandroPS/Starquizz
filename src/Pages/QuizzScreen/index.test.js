import React from 'react';
import ReactDOM from 'react-dom';
import { create } from "react-test-renderer";
import { shallow, mount } from "enzyme";
import QuizzScreen from './';

describe('Quizz Screen', () => {
    const props = {
        characters: [
            {name: 'Luke Skywalker'},
            {name: 'Darth Vader'}
        ],
        guessedCharacters: [],
        onNextPage: ()=>{},
        onPreviousPage: ()=>{},
        getResource: ()=>{}
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<QuizzScreen {...props}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
        const tree = create(<QuizzScreen {...props}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('shows cards', () => {
        const component = mount(<QuizzScreen {...props}/>);

        expect(component.find(".cards-wrapper .card").length).toEqual(2);
    });

});
