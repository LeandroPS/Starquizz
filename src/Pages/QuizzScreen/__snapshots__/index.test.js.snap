// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quizz Screen renders correctly 1`] = `
<div
  className="quizz-screen"
>
  <div
    className="header"
  >
    <div
      className="container"
    >
      <img
        alt="Cabeça do Darth Vader"
        src="darth.png"
      />
      <div
        className="timer"
      >
        02:00
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="cards-wrapper"
    >
      <div
        className="card false"
      >
        <div
          className="thumb"
          style={
            Object {
              "backgroundImage": "url(null)",
            }
          }
        >
          <button
            onClick={[Function]}
          >
            <i
              className="fas fa-question"
            />
          </button>
        </div>
        <div
          className="answer false"
        >
          <input
            disabled={false}
            onChange={[Function]}
            placeholder="Resposta"
            type="text"
            value=""
          />
          <button
            onClick={[Function]}
          >
            Ok
          </button>
        </div>
      </div>
      <div
        className="card false"
      >
        <div
          className="thumb"
          style={
            Object {
              "backgroundImage": "url(null)",
            }
          }
        >
          <button
            onClick={[Function]}
          >
            <i
              className="fas fa-question"
            />
          </button>
        </div>
        <div
          className="answer false"
        >
          <input
            disabled={false}
            onChange={[Function]}
            placeholder="Resposta"
            type="text"
            value=""
          />
          <button
            onClick={[Function]}
          >
            Ok
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="navigation-buttons"
  >
    <button
      onClick={[Function]}
    >
      Anterior
    </button>
    <button
      onClick={[Function]}
    >
      Próximo
    </button>
  </div>
  <div
    className="feedback-window false"
  >
    <div
      className="centered"
    >
      <h1>
        Quizz finalizado
      </h1>
      <div
        className="points"
      >
        <span
          className="title"
        >
          você fez
        </span>
        <span
          className="points"
        >
          0
           Pontos!
        </span>
      </div>
      <form
        onSubmit={[Function]}
      >
        <span>
          Preencha este formulário pra salvar a sua pontuação:
        </span>
        <input
          className="name"
          onChange={[Function]}
          placeholder="Nome"
          required={true}
          type="text"
        />
        <input
          className="email"
          onChange={[Function]}
          placeholder="E-mail"
          required={true}
          type="email"
        />
        <button
          className="button"
        >
          Salvar
        </button>
      </form>
    </div>
  </div>
</div>
`;
